pinot-spi/src/main/java/org/apache/pinot/spi/utils/ByteArray.java

```diff
@@ -31,6 +34,21 @@
  * </ul>
  */
 public class ByteArray implements Comparable<ByteArray>, Serializable {
+
+  private static final MethodHandle COMPARE_UNSIGNED;
+
+  static {
+    MethodHandle compareUnsigned = null;
+    try {
+      compareUnsigned = MethodHandles.publicLookup().findStatic(Arrays.class, "compareUnsigned",
+          MethodType.methodType(int.class,
+              byte[].class, int.class, int.class,
+              byte[].class, int.class, int.class));
+    } catch (NoSuchMethodException | IllegalAccessException ignore) {
```

Comment:
will be good to log a warning that we cannot apply vectorization, I am guessing this would happen with jdk8 or some security restriction?

Reply:
Only on JDK8 AFAIK, but let's log it anyway